core/src/main/java/io/confluent/rest/ApplicationServer.java

```diff
@@ -391,6 +392,9 @@ private void configureConnectors(SslContextFactory sslContextFactory) {
     final HttpConfiguration httpConfiguration = new HttpConfiguration();
     httpConfiguration.setSendServerVersion(false);
 
+    // we need to set below only if listener is https
+    httpConfiguration.addCustomizer(new SecureRequestCustomizer());
```

Comment:
I think we may want to set if `https` listener in configured
```suggestion
    //moved line 405-406 to here
    @SuppressWarnings("deprecation")
    List<URI> listeners = parseListeners(config.getList(RestConfig.LISTENERS_CONFIG),
            config.getInt(RestConfig.PORT_CONFIG), Arrays.asList("http", "https"), "http");
            
    for (URI listener : listeners) {
      if (listener.getScheme().equals("https")) {
        if (httpConfiguration.getCustomizer(SecureRequestCustomizer.class) == null)
          httpConfiguration.addCustomizer(new SecureRequestCustomizer());
      }
    }

Reply:
Good point. Pushed an update along the lines of your suggestion. Should've paid more attention in the first place :)